[ARM][AArch64] Enable FEAT_FHM for Arm Neoverse N2

jthackray opened the pull request:
Correct an issue with Arm Neoverse N2 after it was
changed to a v9a core in change f576cbe44eabb8a5ac0af817424a0d1e7c8fbf85:

  * FEAT_FHM should be enabled for this core.

jthackray committed:
[ARM][AArch64] Enable FEAT_FHM, disable FEAT_MEC for Arm Neoverse N2

Correct a couple of issues with Arm Neoverse N2 after it was
changed to a v9a core in change f576cbe44eabb8a5ac0af817424a0d1e7c8fbf85:

 * FEAT_FHM should be enabled for this core.
 * FEAT_MEC should be disabled for this core, since this is an
   optional Armv9.2-A and later feature

efriedma-quic commented on llvm/lib/Target/AArch64/AArch64.td:
```diff
@@ -684,7 +684,7 @@ def HasV8_9aOps : SubtargetFeature<
 def HasV9_0aOps : SubtargetFeature<
   "v9a", "HasV9_0aOps", "true", "Support ARM v9a instructions",
-  [HasV8_5aOps, FeatureMEC, FeatureSVE2]>;
+  [HasV8_5aOps, FeatureSVE2]>;
```
This looks like it affects every CPU that declares support for armv9; is this correct for all of them?

jthackray replied:
I will update this patch to leave FEAT_MEC unchanged for now, but we will probably (in future) disable it for all CPUs (and add a flag to enable), since it's a v9.2a and higher optional feature.

jthackray committed:
amend! [ARM][AArch64] Enable FEAT_FHM for Arm Neoverse N2

Correct an issue with Arm Neoverse N2 after it was
changed to a v9a core in change f576cbe44eabb8a5ac0af817424a0d1e7c8fbf85:

* FEAT_FHM should be enabled for this core.

vhscampos commented:
Please double check if the arch features are listed in sorted alphabetical order. If so, please put FHM and fp16fml in their correct places.

jthackray committed:
fixup! [ARM][AArch64] [ARM][AArch64] Enable FEAT_FHM for Arm Neoverse N2

jthackray commented:
> Please double check if the arch features should listed in sorted alphabetical order. If so, please put FHM and fp16fml in their correct places.

Sure, now ordered when in a sorted list (the lists in TargetParserTest.cpp are not already sorted).

vhscampos commented:
Looks good to me, but please wait for the others' approval. Thank you

rgwott reviewed (approved):
LGTM.
